### ISSUE TITLE
Legacy HTML Support

### ISSUE THREAD

**Issue by tresf** [qzind/tray#277]
It has been brought to our attention that some HTML content still prints better in 1.9 versus 2.1.

Add the 1.8/1.9 HTML printing support back into 2.0 branch which will help some clients finally make the switch to a supported version.

Download here: https://github.com/tresf/tray/releases/tag/v2.0.6

https://qz.io/wiki/2.0-Pixel-Printing#legacy-html

```js
config = qz.configs.create("PrinterName", { legacy: true });
   printData = [{
   type: 'html',
   format: 'file',
   data: 'https://foo.bar/file.html'
}];

qz.print(config, printData);
```

**Comment by bberenz** [comment]
Feature ready for testing under `legacy-html` branch

**Comment by klabarge** [comment]
Using `legacy: "true"`, only the file path to the HTML page is printed.

Switching off of `legacy` yields expected output.

![image](https://user-images.githubusercontent.com/12505463/36770020-e238d080-1c15-11e8-82a0-44a7c2d632f2.png)

**Comment by bberenz** [comment]
Looks like I forgot to add some file support. There is a basic support under 595b250b85e44215398d954213f38c5dcf8705b3. Be aware that it appears swing cannot handle newer HTML standards, so this isn't perfect.

**Comment by tresf** [comment]
@bberenz thanks that fixed it.  The only issue is that for some reason, the receipts are about 1/4 inch too far to the left.  Adjusting the margins on the printer don't seem to do anything
**Actual** &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; **Desired**
![screen shot 2018-03-01 at 1 22 57 am](https://user-images.githubusercontent.com/6345473/36829885-1d214acc-1cef-11e8-9f4a-4b3f4acf3502.png) ![screen shot 2018-03-01 at 1 37 29 am](https://user-images.githubusercontent.com/6345473/36830345-2816d3b4-1cf1-11e8-80ff-d715f53cf99d.png)

@lite1979 may know a bit more, but from the above picture, you can see that when I use the XPS Document writer it does visually show the same receipt being cut off the left margin, so this may a configuration on the PC or it may be a slight change in the utilization of the imageable area between 1.9 and 2.9.

**Comment by tresf** [comment]
I was able to tackle the margin problems by adjusting the style tags.  This is working well.

**Comment by klabarge** [comment]
There appears to be a margin issue with legacy HTML printing on macOS. The top and left margins are cut off. This is not reproducible on Windows or Ubuntu.

For baseline, I printed the same HTML through 1.9 on macOS. The margins in 1.9 are cutoff on the bottom right.

![image](https://user-images.githubusercontent.com/12505463/37943904-9fc50ace-3147-11e8-8bee-1cf1cda6ec1d.png)

**Comment by tresf** [comment]
MacOS bug patched via 86af2ba, ready for testing. :)

**Comment by klabarge** [comment]
Patch worked, and I regression tested against Ubuntu and Windows 👍 

On the left is legacy-html, on the right is normal.

![image](https://user-images.githubusercontent.com/12505463/38146301-6c6e0136-341b-11e8-8e0e-bf6d30e8a715.png)
